src/js/tracker.js

```diff
@@ -796,6 +845,220 @@
+				if (visitor.hasOwnProperty('referrer')) context['referrer'] = visitor.referrer;
+				if (visitor.hasOwnProperty('os')) context['os'] = visitor.os;
+
+				return {
+					schema: 'iglu:com.optimizely/visitor/jsonschema/1-0-0',
+					data: context
+				};
+			}
+		}
+
+		/**
+		 * Creates a context from the window['optimizely'].data.visitor.audiences object
+		 *
+		 * @return array VisitorAudience contexts
+		 */
+		function getOptimizelyAudienceContexts() {
+			var audienceIds = window['optimizely'].data.visitor.audiences;
+			if (audienceIds) {
+				var contexts = [];
+
+				for (var key in audienceIds) {
+					var context = {};
+					context['id'] = key;
+					context['isMember'] = audienceIds[key]; 
+
+					contexts.push({
+						schema: 'iglu:com.optimizely/visitor_audience/jsonschema/1-0-0',
+						data: context
+					});
+				}
+				return contexts;
+			}
+			return [];
+		}
+
+		/**
+		 * Creates a context from the window['optimizely'].data.visitor.dimensions object
+		 *
+		 * @return array VisitorDimension contexts
+		 */
+		function getOptimizelyDimensionContexts() {
+			var dimensionIds = window['optimizely'].data.visitor.dimensions;
+			if (dimensionIds) {
+				var contexts = [];
+
+				for (var key in dimensionIds) {
+					var context = {};
+					context['id'] = key;
+					context['value'] = dimensionIds[key]; 
+
+					contexts.push({
+						schema: 'iglu:com.optimizely/visitor_dimension/jsonschema/1-0-0',
+						data: context
+					});
+				}
+				return contexts;
+			}
+			return [];
+		}
+
+		/**
+		 * Creates a context from the window['augur'] object
+		 *
+		 * @return object The IdentityLite context
+		 */
+		function getAugurIdentityLiteContext() {
+			var augur = window['augur'];
+			if (augur) {
+				var context = { consumer: {}, device: {} };
+				var consumer = augur.hasOwnProperty('consumer') ? augur.consumer : {};
+				if (consumer.hasOwnProperty('UID')) context['consumer']['UUID'] = consumer.UID;
+				var device = augur.hasOwnProperty('device') ? augur.device : {};
+				if (device.hasOwnProperty('ID')) context['device']['ID'] = device.ID;
+				if (device.hasOwnProperty('isBot')) context['device']['isBot'] = device.isBot;
+				if (device.hasOwnProperty('isProxied')) context['device']['isProxied'] = device.isProxied;
+				if (device.hasOwnProperty('isTor')) context['device']['isTor'] = device.isTor;
+
+				// isIncognito can be found in two places
+				if (device.hasOwnProperty('isIncognito')) context['device']['isIncognito'] = device.isIncognito;
+				if (!context.device.hasOwnProperty('isIncognito')) {
+					var fingerprint = device.hasOwnProperty('fingerprint') ? device.fingerprint : {};
+					if (fingerprint.hasOwnProperty('browserHasIncognitoEnabled')) {
+						context['device']['isIncognito'] = fingerprint.browserHasIncognitoEnabled;
+					}
+				}
+
+				return {
+					schema: 'iglu:io.augur.snowplow/identity_lite/jsonschema/1-0-0',
+					data: context
+				};
+			}
+		}
```

Comment:
In JS trying to get a nonexistent property of an object doesn't throw an exception, it just returns `undefined`. It can make debugging hard but in this case it is useful.

Note that this doesn't work for nested properties:

```js
({}).device; // undefined
({}).device.isIncognito; // Uncaught TypeError: Cannot read property 'isIncognito' of undefined
```

To do it with nested properties, you need [brototype](https://github.com/letsgetrandy/brototype).